util/daghash/hash_test.go

```diff
@@ -425,3 +426,40 @@ func TestSort(t *testing.T) {
 		}
 	}
 }
+
+func hashFlipBit(hash Hash, bit int) Hash {
+	word := bit / 8
+	bit = bit % 8
+	hash[word] ^= 1 << bit
+	return hash
+}
+
+func TestHash_Cmp(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+
+	for i := 0; i < 100; i++ {
+		hash := Hash{}
+		n, err := r.Read(hash[:])
+		if err != nil || n != len(hash) {
+			t.Fatalf("Failed generating a random hash '%s'", err)
```

Comment:
err is not necessarily set here. Please split err != nil and n != len(hash) into two conditions.

Reply:
Done